joomla/joomla-cms
# [5.3] Control access to component preferences individually

Pull Request for Issue # .

### Summary of Changes
Modified code in administrator com_config view for component to allow fine control of preferences to be presented to users.

Currently full access is granted to any user having admin access to a component.
This change allows access to be granted individually to fieldsets in a component's access.xml

### Testing Instructions
Create a custom component and set up access actions in its access.xml corresponding to fieldsets in its config.xml, 
eg, in access xml create more than one fields of the form

	<action name="core.options.pref1" title="Preference 1" description="Show preference 1" />

corresponding to fields in config.xml like

    <fieldset
		name="pref1"
		label="Preference 1"
	>
        <field
            name='something'
            type='text'
            label='Label'
            class='inputbox'
        />
    </fieldset>

Create a view in the component with a toolbar with the options button in it with the usual access control.
Install the component and go to its permissions tab in preferences.
Allow admin access to this new component for a non-superuser.
Log in with this user and see that the component's preference are available to the user, but only the permissions tab.
Looking at the permissions tab notice the new options actions at the bottom of the list.  Enable one and redisplay the options for the selected user and see that selected tabs are now available.

Change and save an option on a (non-permissions) tab available to a non-superuser and then check that preference values on tabs not available to that user remain intact.

### Actual result BEFORE applying this Pull Request

Access to component preferences cannot be controlled individually.

### Expected result AFTER applying this Pull Request

Access to component preferences **can** be controlled individually.

### Link to documentations
I'm happy to write documentation for this change should it be accepted.

Please select:
